aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,104 @@
 ---
-title: 🔧 Response caching middleware
+title: Response Caching Middleware | Microsoft Docs
 author: rick-anderson
 ms.author: riande
 manager: wpickett
-ms.date: 10/14/2016
+ms.date: 12/05/2016
 ms.topic: article
 ms.assetid: f9267eab-2762-42ac-1638-4a25d2c9d67c
 ms.prod: aspnet-core
 uid: performance/caching/middleware
 ---
-# 🔧 Response caching middleware
+# Response Caching Middleware
+
+By [Luke Latham](https://github.com/GuardRex) and [John Luo](https://github.com/JunTaoLuo)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/caching/response/sample)
+
+This document provides details on how to configure the Response Caching Middleware for ASP.NET Core applications. For an introduction to to HTTP caching and the `ResponseCaching` feature, see [Response Caching](response.md).
+
+## Package
+To include the middleware in your project, add a reference to the  [`Microsoft.AspNetCore.ResponseCaching`](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package. The middleware is available for projects that target `.NETFramework 4.5.1` or `.NETStandard 1.3` or higher.
+
+## Extensions
+In `ConfgureServices`, add the middleware to your service collection.
+```c#
+public void ConfigureServices(IServiceCollection services)
+{
+    ...
+    services.AddResponseCaching();
+    ...
+}
+```
+Configure the application to use the middleware when processing requests.
+```c#
+public void Configure(IApplicationBuilder app)
+{
+    ...
+    app.UseResponseCaching();
+    ...
+}
+```
+The position of `app.UseResponseCaching()` relative to other middleware in the pipeline is important. Any terminal middleware placed before the Response Caching Middleware will prevent the Response Caching Middleware from caching or serving the response. For example, if you place [Static File Middleware](xref:fundamentals/static-files) before `app.UseResponseCaching()`, your static files will not be cached by the middleware. If you place Static File Middleware after `app.UseResponseCaching()`, your static files will be cached.
```

Comment:
@Rick-Anderson Yep ... it's all over the map. Just for Static File Middleware, I see ...
* static file middleware
* StaticFileMiddleware
* static files middleware
* Static File middleware

Would you like to see a PR that cleans up middleware names? If so, would you like title case for full, explicit middleware names?